vassal-app/src/main/java/VASSAL/build/module/map/Flare.java

```diff
@@ -0,0 +1,343 @@
+  private int frames;
+  private int framesPerPulse;
+  private String flareKey;
+  private Color color;
+  private static int CIRCLE_RATE    = 33; // 33ms for approx 30 frames/sec
+  private static int DEFAULT_FRAMES = 60; // 2 sec for basic "no animation" pulse
+  private Point clickPoint;
+  private Boolean active;
+  private Timer timer;
+  public static final String CIRCLE_SIZE  = "circleSize";
+  public static final String CIRCLE_COLOR = "circleColor";
+  public static final String FLARE_KEY    = "flareKey";
+  public static final String PULSES       = "flarePulses";
+  public static final String PULSES_PER_SEC = "flarePulsesPerSec";
+  
+  public static final String FLARE_ALT_LOCAL     = Resources.getString("Editor.Flare.flare_key_desc", Resources.getString("Keys.alt"));
+  public static final String FLARE_CTRL_LOCAL    = Resources.getString("Editor.Flare.flare_key_desc", Resources.getString("Keys.ctrl"));
+  public static final String FLARE_COMMAND_LOCAL = Resources.getString("Editor.Flare.flare_key_desc", Resources.getString("Keys.meta"));
+  
+  public static final String FLARE_ALT = "keyAlt";
+  public static final String FLARE_CTRL = "keyCtrl";
+  
+
+
+  public Flare() {
+    this.circleSize = 100;
+    this.color = Color.RED;
+    this.active = false;
+    this.flareKey = FLARE_ALT;
+    this.pulses       = 6;
+    this.pulsesPerSec = 3;
+  }
+
+  public String getDescription() {
+    return Resources.getString("Map Flare");
+  }
+  
+  public static String getConfigureTypeName() {
+    return Resources.getString("Map Flare");
+  }
+
+  public Class<?>[] getAttributeTypes() {
+    return (Class<?>[]) new Class[] { FlareKeyConfig.class, Integer.class, Color.class, Integer.class, Integer.class };
+  }
+
+  public String[] getAttributeNames() {
+    return new String[] { FLARE_KEY, CIRCLE_SIZE, CIRCLE_COLOR, PULSES, PULSES_PER_SEC };
+  }
+
+  public String[] getAttributeDescriptions() {
+    return new String[] { Resources.getString("Editor.Flare.flare_key"), 
+                          Resources.getString("Editor.Flare.circle_size"), 
+                          Resources.getString("Editor.Flare.circle_color"),
+                          Resources.getString("Editor.Flare.pulses"),
+                          Resources.getString("Editor.Flare.pulses_per_sec") };
+  }
+
+  public String getAttributeValueString(final String key) {
+    if (FLARE_KEY.equals(key)) {
+      return new StringBuilder().append(this.flareKey).toString();
+    }
+    else if (CIRCLE_SIZE.equals(key)) {
+      return new StringBuilder().append(this.circleSize).toString();
+    }
+    else if (CIRCLE_COLOR.equals(key)) {
+      return new StringBuilder().append(ColorConfigurer.colorToString(this.color)).toString();
+    }
+    else if (PULSES.equals(key)) {
+      return new StringBuilder().append(this.pulses).toString();
+    }
+    else if (PULSES_PER_SEC.equals(key)) {
+      return new StringBuilder().append(this.pulsesPerSec).toString();
+    }
```

Comment:
For all of these, you're not doing any concatenation, so you can just return `String.valueOf()` for the numbers, and `ColorConfigurer.colorToString(this.color)` directly.

Reply:
fixed